### PR TITLE
feat: add release workflow to build and publish a container

### PR DESCRIPTION
**Pull request by EliseZeroTwo**
This PR adds a `release.yml` that contains a workflow that runs when a release is published. It creates and publishes an image for the tag on the GHCR.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Publish Container to Packages
+
+on:
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
```
Comment (chgrv) [inline .github/workflows/release.yml]: As I understand it we should use Nexus as our container registry across the board now, so the registry for this should be `docker-oss.nexus.famedly.de`. See [this notion page](https://www.notion.so/famedly/Nexus-Registry-55dfc4abe7ef4a8ea03946d655238b5d) for more info. I will see if we can make a post in the announcements room about this to make sure everyone is made aware.

You can authenticate to `docker{-oss}.nexus.famedly.de` with the user `famedly-ci` and password stored in `secrets.REGISTRY_PASSWORD`.

**Review by emgrav — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (EliseZeroTwo) [reply .github/workflows/release.yml]: Thank you, I have changed this now

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+on:
+  release:
+    types: [published]
+
+env:
+  REGISTRY: docker-oss.nexus.famedly.de
+  REGISTRY_USERNAME: famedly-ci
+  REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+  IMAGE_NAME: ${{ github.event.repository.name }}
```
Comment (chgrv) [inline .github/workflows/release.yml:11]: Aiui this result in the image name being `Meow-Coverage`, right? If we want naming consistency with our other images it should probably be lowercased to `meow-coverage`.

**Review by emgrav — APPROVED** [review]
Looks good! I noticed one more thing about the naming that I missed in the first review, but I could be wrong about it, if I am just go ahead and merge it.

File: .github/workflows/release.yml (hunk 2 above)
Comment (EliseZeroTwo) [reply .github/workflows/release.yml:11]: Yes, it should, nice catch! I fixed this now with a step that converts the environment variable to lowercase at the start.

**Review by emgrav — APPROVED** [review]

**Pull request merged**